src/libs/ValidationUtils.ts

```diff
@@ -265,6 +266,12 @@ function isValidUSPhone(phoneNumber = '', isCountryCodeOptional?: boolean): bool
     const phone = phoneNumber || '';
     const regionCode = isCountryCodeOptional ? CONST.COUNTRY.US : undefined;
 
+    // When we pass regionCode as an option to parsePhoneNumber it wrongly assumes inputs like '=15123456789' as valid
+    // so we need to check if it is a valid phone.
+    if (regionCode && !Str.isValidPhone(phone)) {
```

Comment:
This change stopped supporting the other valid number formats and caused an issue https://github.com/Expensify/App/issues/37723